src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1175,6 +1185,10 @@ public int updateCertificatesStatusScheduled() {
     // Internal Use Only
     public void updateCertificateUser(UUID certificateUuid, String userUuid) throws NotFoundException {
         Certificate certificate = certificateRepository.findByUuid(certificateUuid).orElseThrow(() -> new NotFoundException(Certificate.class, certificateUuid));
+        if (certificate.isArchived()) {
+            logger.info("Certificate with UUID {} is archived and the user cannot be set.", certificateUuid);
```

Comment:
Here is more suitable to throw Exception since it is user triggered operation that is not allowed. It should say it is not allowed to set archived certificate for user as auth method.